### ISSUE TITLE
the example of coupon code doesn' work from the 3.0.11 version

### ISSUE THREAD

**Issue by luarmr** [recurly/recurly-js#193]
The next code extract from the examples in https://docs.recurly.com/js/#pricing-api
```
var pricing = recurly.Pricing();

pricing
  .plan('basic', { quantity: 1 })
  .coupon('coop')
  .catch(function (err) {
    // err.code
  })
  .done(function (price) {
    // price object as emitted by 'change' event
  });
```
The error code is not executing when the coupon is failing.

I go back to the previous version and i write this for alert the people have the same problem, because i didn't search inside the code at the moment. If i will try to collect more info.

**Comment by chrissrogers** [comment]
You are correct that the .catch will not catch coupon not found errors. This is due to the line changed in your PR; however, the line should remain, as it is not our intention to throw an error when a coupon is not found, and instead to fail silently while removing all coupons from the pricing object.

I think the best solution here is to emit an invalid coupon event, for which we have a discussion in #176

**Comment by chrissrogers** [comment]
In versions previous to `3.0.11`, we were mistakenly emitting an error. This was fixed in #177
